### PR TITLE
Change exceptions

### PR DESCRIPTION
**Pull request by Feazom**

### CODE REVIEW

File: open_schools_platform/api/errors.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        return list(map(selector, detail))
+    return [selector(detail), ]
+
+
+def expand_nested(dictionary: dict):
+    result: Dict[str, Any] = {}
+    for key, value in dictionary.items():
+        if isinstance(value, dict):
+            temp = expand_nested(value)
+            result = dict(result, **dict([(f'{key}.{i}', temp[i]) for i in temp if temp[i] is not None]))
```
Comment (zavarin-michael) [inline open_schools_platform/api/errors.py:60]: why we need to expand nested?
it will be less convinient if we use nested dicts?

**Review by zavarin-michael — COMMENTED** [review]
Explanations, please)

File: open_schools_platform/api/errors.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if not isinstance(violations, list):
+        violations = [violations]
+    violations = None if len(violations) == 0 else violations
+
+    message = None
+    if isinstance(exception.detail, dict):
+        message = ';\n'.join(
+            map(lambda item: f"'{item[0]}': {','.join(item[1])}", flatten(process_detail(exception.detail).items())))
+    else:
+        message = '\n'.join(flatten(process_detail(exception.detail)))
```
[the remark this answers is not in the document]
Comment (Feazom) [reply open_schools_platform/api/errors.py:40]: It's supposed to convert the list [[1, 2], [3], 4] to [1, 2, 3, 4] but I found that this is redundant now

File: open_schools_platform/api/errors.py (hunk 1 above)
Comment (Feazom) [reply open_schools_platform/api/errors.py:60]: I did this in order to use one versatile serializer

Comment (zavarin-michael) [reply open_schools_platform/api/errors.py]: ok, understand

**Pull request merged**